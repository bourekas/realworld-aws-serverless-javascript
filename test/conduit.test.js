const { App } = require('aws-cdk-lib');
const { Template } = require('aws-cdk-lib/assertions');
const { ConduitStack } = require('../lib/conduit-stack');

it('does not alter the infrastructure', () => {
  const app = new App();
  const stack = new ConduitStack(app, 'conduit-test-stack');
  const template = Template.fromStack(stack);
  
  expect(template.toJSON()).toMatchSnapshot();
});
