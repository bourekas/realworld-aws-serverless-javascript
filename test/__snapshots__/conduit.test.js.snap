// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`does not alter the infrastructure 1`] = `
{
  "Outputs": {
    "conduitapprestapiEndpoint7FF64387": {
      "Value": {
        "Fn::Join": [
          "",
          [
            "https://",
            {
              "Ref": "conduitapprestapi272CB4F8",
            },
            ".execute-api.",
            {
              "Ref": "AWS::Region",
            },
            ".",
            {
              "Ref": "AWS::URLSuffix",
            },
            "/",
            {
              "Ref": "conduitapprestapiDeploymentStageprodCDA258D9",
            },
            "/",
          ],
        ],
      },
    },
  },
  "Parameters": {
    "BootstrapVersion": {
      "Default": "/cdk-bootstrap/hnb659fds/version",
      "Description": "Version of the CDK Bootstrap resources in this environment, automatically retrieved from SSM Parameter Store. [cdk:skip]",
      "Type": "AWS::SSM::Parameter::Value<String>",
    },
  },
  "Resources": {
    "conduitapprestapi272CB4F8": {
      "Properties": {
        "Name": "rest-api",
      },
      "Type": "AWS::ApiGateway::RestApi",
    },
    "conduitapprestapiANY1851DA47": {
      "Properties": {
        "AuthorizationType": "NONE",
        "HttpMethod": "ANY",
        "Integration": {
          "Type": "MOCK",
        },
        "ResourceId": {
          "Fn::GetAtt": [
            "conduitapprestapi272CB4F8",
            "RootResourceId",
          ],
        },
        "RestApiId": {
          "Ref": "conduitapprestapi272CB4F8",
        },
      },
      "Type": "AWS::ApiGateway::Method",
    },
    "conduitapprestapiAccount52B5AE10": {
      "DeletionPolicy": "Retain",
      "DependsOn": [
        "conduitapprestapi272CB4F8",
      ],
      "Properties": {
        "CloudWatchRoleArn": {
          "Fn::GetAtt": [
            "conduitapprestapiCloudWatchRole08BF49F2",
            "Arn",
          ],
        },
      },
      "Type": "AWS::ApiGateway::Account",
      "UpdateReplacePolicy": "Retain",
    },
    "conduitapprestapiCloudWatchRole08BF49F2": {
      "DeletionPolicy": "Retain",
      "Properties": {
        "AssumeRolePolicyDocument": {
          "Statement": [
            {
              "Action": "sts:AssumeRole",
              "Effect": "Allow",
              "Principal": {
                "Service": "apigateway.amazonaws.com",
              },
            },
          ],
          "Version": "2012-10-17",
        },
        "ManagedPolicyArns": [
          {
            "Fn::Join": [
              "",
              [
                "arn:",
                {
                  "Ref": "AWS::Partition",
                },
                ":iam::aws:policy/service-role/AmazonAPIGatewayPushToCloudWatchLogs",
              ],
            ],
          },
        ],
      },
      "Type": "AWS::IAM::Role",
      "UpdateReplacePolicy": "Retain",
    },
    "conduitapprestapiDeployment1D4BDA332c6fc89dcea4b058ff9b4282d47876fe": {
      "DependsOn": [
        "conduitapprestapiANY1851DA47",
      ],
      "Properties": {
        "Description": "Automatically created by the RestApi construct",
        "RestApiId": {
          "Ref": "conduitapprestapi272CB4F8",
        },
      },
      "Type": "AWS::ApiGateway::Deployment",
    },
    "conduitapprestapiDeploymentStageprodCDA258D9": {
      "DependsOn": [
        "conduitapprestapiAccount52B5AE10",
      ],
      "Properties": {
        "DeploymentId": {
          "Ref": "conduitapprestapiDeployment1D4BDA332c6fc89dcea4b058ff9b4282d47876fe",
        },
        "RestApiId": {
          "Ref": "conduitapprestapi272CB4F8",
        },
        "StageName": "prod",
      },
      "Type": "AWS::ApiGateway::Stage",
    },
  },
  "Rules": {
    "CheckBootstrapVersion": {
      "Assertions": [
        {
          "Assert": {
            "Fn::Not": [
              {
                "Fn::Contains": [
                  [
                    "1",
                    "2",
                    "3",
                    "4",
                    "5",
                  ],
                  {
                    "Ref": "BootstrapVersion",
                  },
                ],
              },
            ],
          },
          "AssertDescription": "CDK bootstrap stack version 6 required. Please run 'cdk bootstrap' with a recent version of the CDK CLI.",
        },
      ],
    },
  },
}
`;
